server/src/jobs/auto-discovery.cpp

```diff
@@ -0,0 +1,275 @@
+*/
+
+#include "auto-discovery.h"
+#include "protocols.h"
+#include "assets.h"
+#include <asset/asset-manager.h>
+#include <fty_asset_dto.h>
+
+namespace fty::job {
+
+Expected<void> AutoDiscovery::updateExt(const commands::assets::Ext &ext_in, fty::asset::create::Ext &ext_out)
+{
+    // Construct and update output ext attributes according input ext attributes
+    // [{ "key": "val", "read_only": "true" },...] -> {
+    //                                                  "key:" {
+    //                                                    "value": "val",
+    //                                                    "readOnly: true,
+    //                                                    "update": true
+    //                                                  },
+    //                                                  ...
+    //                                                }
+    for (auto it = ext_in.begin(); it != ext_in.end(); it++) {
+        fty::asset::create::MapExtObj newMapExtObj;
+        std::string key;
+        // for each attribute
+        for (auto it_attr = it->begin(); it_attr != it->end(); it_attr++) {
+            if (it_attr->first == "read_only") {
+                newMapExtObj.readOnly = it_attr->second == "true" ? true : false;
+            }
+            else {
+                // It is the name of the key and its associated value
+                key = it_attr->first;
+                newMapExtObj.value = it_attr->second;
+            }
+        }
+        if (!key.empty()) {
+            ext_out.append(key, newMapExtObj);
+        }
+    }
+    return {};
+}
+
+Expected<void> AutoDiscovery::updateHostName(const std::string& address, fty::asset::create::Ext &ext)
+{
+    if (!address.empty()) {
+        // Set host name
+        struct sockaddr_in sa_in;
+        struct sockaddr* sa  = reinterpret_cast<sockaddr*>(&sa_in);
+        socklen_t len = sizeof(sockaddr_in);
+        char dns_name[NI_MAXHOST];
+        sa_in.sin_family = AF_INET;
+        memset(dns_name, 0, NI_MAXHOST);
+        if (inet_aton(address.c_str(), &sa_in.sin_addr) == 1) {
+            if (!getnameinfo(sa, len, dns_name, sizeof(dns_name), NULL, 0, NI_NAMEREQD)) {
+                auto &it_ext_dns = ext.append("dns.1");
+                it_ext_dns.value = dns_name;
+                it_ext_dns.readOnly = false;
+                logDebug("Retrieved DNS information: FQDN = '{}'", dns_name);
+                char* p = strchr(dns_name, '.');
+                if (p) {
+                    *p = 0;
+                }
+                auto &it_ext_hostname = ext.append("hostname");
+                it_ext_hostname.value = dns_name;
+                it_ext_hostname.readOnly = false;
+                logDebug("Hostname = '{}'", dns_name);
+            }
+            else {
+                return fty::unexpected("No host information retrieved from DNS");
```

Comment:
clang format and camel cases upper, and please provide more info in error with help of strerror